test cases/common/212 dependency get_variable method/meson.build

```diff
@@ -62,3 +62,4 @@ assert(idep.get_variable(pkgconfig : 'foo', cmake : 'foo', configtool : 'foo',
 
 idep = declare_dependency(variables : ['foo=value'])
 assert(idep.get_variable(internal: 'foo') == 'value')
+assert(idep.get_variable('foo') == 'value')
```

Comment:
I would however like some tests that cover a case like `dep.get_variable('foo', cmake : 'FOO')`, (or pkg-config, same idea) where the specialized variable both is and isn't found, to ensure that the specialized variable has precedence.

Reply:
On the previous line we could change to:
```
assert(idep.get_variable('FOO', internal: 'foo') == 'value')
```

Reply:
@eli-schwartz Added that test too, good point.

Reply:
Please do leave the just `get_variable(internal : 'foo')` we should have a test with and without the default.

Reply:
Sure, I now test the 3 cases.